apis/openstack-nova/src/test/java/org/jclouds/openstack/nova/v2_0/extensions/HypervisorApiExpectTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.openstack.nova.v2_0.extensions;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableSet;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.http.HttpResponse;
+import org.jclouds.openstack.nova.v2_0.NovaApi;
+import org.jclouds.openstack.nova.v2_0.domain.regionscoped.Hypervisor;
+import org.jclouds.openstack.nova.v2_0.domain.regionscoped.HypervisorDetails;
+import org.jclouds.openstack.nova.v2_0.internal.BaseNovaApiExpectTest;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "unit", testName = "HypervisorApiExpectTest")
+public class HypervisorApiExpectTest extends BaseNovaApiExpectTest {
+
+   public void testWhenNamespaceInExtensionsListHypervisorPresent() throws Exception {
```

Comment:
?Hmm yes, I like the pattern actually. I want to run it through some tests, but looks good. If it works ok, we might have to refactor some other mock tests to use it as well.

________________________________
From: arvindn05 <notifications@github.com>
Sent: Thursday, July 23, 2015 3:51 AM
To: jclouds/jclouds
Cc: Zack Shoylev
Subject: Re: [jclouds] add os-hypervisors extension (#819)


In apis/openstack-nova/src/test/java/org/jclouds/openstack/nova/v2_0/extensions/HypervisorApiExpectTest.java<https://github.com/jclouds/jclouds/pull/819#discussion_r35301354>:

> +import org.jclouds.http.HttpRequest;
> +import org.jclouds.http.HttpResponse;
> +import org.jclouds.openstack.nova.v2_0.NovaApi;
> +import org.jclouds.openstack.nova.v2_0.domain.regionscoped.Hypervisor;
> +import org.jclouds.openstack.nova.v2_0.domain.regionscoped.HypervisorDetails;
> +import org.jclouds.openstack.nova.v2_0.internal.BaseNovaApiExpectTest;
> +import org.testng.annotations.Test;
> +
> +import static org.testng.Assert.assertEquals;
> +import static org.testng.Assert.assertFalse;
> +import static org.testng.Assert.assertTrue;
> +
> +@Test(groups = "unit", testName = "HypervisorApiExpectTest")
> +public class HypervisorApiExpectTest extends BaseNovaApiExpectTest {
> +
> +   public void testWhenNamespaceInExtensionsListHypervisorPresent() throws Exception {


figured out the issue. It was because of me using keystoneAuthResponse.json. Somehow the mock tests only work with access.json

Also tried to make use of beforemethod and aftermethod annotations to do some of the setup. since the tests run in parallel, i could not use them directly. so had to get creative to reduce code duplication.
if you like this pattern maybe future tests can use it...

-
Reply to this email directly or view it on GitHub<https://github.com/jclouds/jclouds/pull/819/files#r35301354>.